src/dmd/dcast.d

```diff
@@ -3319,11 +3319,42 @@ Lagain:
         {
             if (t1.ty == Tvector || t2.ty == Tvector)
                 return Lincompatible();
+            auto t1old = e1.type;
+            auto t2old = e2.type;
             e1 = integralPromotions(e1, sc);
             e2 = integralPromotions(e2, sc);
             t1 = e1.type;
             t2 = e2.type;
-            goto Lagain;
+
+            if (t1old != t1 || t2old != t2)
+            {
+                // only try this again if the integral promotion changed anything!
+                goto Lagain;
+            }
+            else
```

Comment:
This doesn't need to be an `else`

Reply:
I am afraid if this is not an else, I cannot isolate the only where integral promotion does not change things ...